proxy/handlers/base.go

```diff
@@ -2,41 +2,50 @@ package handlers
 
 import "github.com/gin-gonic/gin"
 
+// EndpointGroupHandler defines the components of a gin.Engine Group
+// It allows for multiple middleware configuration
+type EndpointGroupHandler struct {
+	Root             string
+	Middleware       []gin.HandlerFunc
```

Comment:
Middlewares

Reply:
plural is also `middleware`. can change so it reflects it's an array.

Reply:
renamed